### PR TITLE
Update the fast parser to work with Python 2.7

### PR DESCRIPTION
**Pull request by ddfisher**
Next step for #1353.

### CODE REVIEW

File: mypy/fastparse.py [hunk 1]
```diff
@@ -21,7 +21,9 @@
 from mypy.errors import Errors
 
 try:
-    import typed_ast  # type: ignore
+    from typed_ast import ast27  # type: ignore
+    from typed_ast import ast35  # type: ignore
+    from typed_ast import conversions  # type: ignore
```
Comment (gvanrossum) [inline mypy/fastparse.py]: Maybe you could provide stubs for these? The untypedness here kind of concerns me.

File: mypy/fastparse.py [hunk 2]
```diff
@@ -43,15 +45,21 @@ def parse(source: Union[str, bytes], fnam: str = None, errors: Errors = None,
     """
     is_stub_file = bool(fnam) and fnam.endswith('.pyi')
     try:
-        ast = typed_ast.parse(source, fnam, 'exec')
+        if pyversion[0] == 3 or is_stub_file:
+            ast = ast35.parse(source, fnam, 'exec')
+        else:
```
Comment (gvanrossum) [inline mypy/fastparse.py]: I'd assert that pyversion[0] == 2 here, or else use >= 3 two lines up (which is what every other pyversion test in mypy uses).

File: mypy/fastparse.py [hunk 3]
```diff
@@ -84,11 +92,14 @@ def find(f: Callable[[T], bool], seq: Sequence[T]) -> T:
     return None
 
 
-class ASTConverter(typed_ast.NodeTransformer):
-    def __init__(self):
-        self.in_class = False
+class ASTConverter(ast35.NodeTransformer):
+    def __init__(self, pyversion, custom_typing_module = None):
```
Comment (gvanrossum) [inline mypy/fastparse.py]: Why no type annotations here?

File: mypy/fastparse.py [hunk 4]
```diff
@@ -84,11 +92,14 @@ def find(f: Callable[[T], bool], seq: Sequence[T]) -> T:
...
+        self.class_nesting = 0
         self.imports = []
 
+        self.pyversion = pyversion
+        self.custom_typing_module = custom_typing_module
+
     def generic_visit(self, node):
```
Comment (gvanrossum) [inline mypy/fastparse.py]: Or here, or in any other method here?

**Comment by gvanrossum** [comment]
Mostly LGTM, would like your response to the review feedback so far.

Generic comment: maybe instead of changing every place that tests for StrExpr to now check for StrExpr or BytesExpr, perhaps those should be the same class with a flag field to distinguish the two? (And why isn't UnicodeExpr also included?)

File: mypy/fastparse.py (hunk 2 above)
Comment (ddfisher) [inline mypy/fastparse.py]: Changed to use `>= 3`.

**Comment by gvanrossum** [comment]
If you rebase, the typeshed sync will be in and then maybe the tests will pass (or give useful info).

**Comment by ddfisher** [comment]
Build passed!

**Comment by ddfisher** [comment]
RE: StrExpr vs BytesExpr, I think it makes sense to keep them as-is because there's a separate Str vs Bytes in the Python 3 ast, which we'll hopefully be switching to soon.

**Pull request merged**